app/components/NodeSelectPanel/NodeSelectPanel.jsx

```diff
@@ -0,0 +1,147 @@
+// @flow
+import React from 'react'
+import classNames from 'classnames'
+
+import { ROUTES } from '../../core/constants'
+import FullHeightPanel from '../Panel/FullHeightPanel'
+import NodeSelectIcon from '../../assets/icons/node-select.svg'
+import CloseButton from '../CloseButton'
+import ConfirmIcon from '../../assets/icons/confirm.svg'
+import RefreshIcon from '../../assets/icons/refresh.svg'
+import AddIcon from '../../assets/icons/add.svg'
+import Tooltip from '../Tooltip'
+import styles from './NodeSelectPanel.scss'
+
+type Node = {
+  latency: string,
+  url: string,
+  blockCount: int
+}
+
+type Props = {
+  nodes: Node[],
+  nodesShown: int,
+  loading: Boolean,
+  loadNodesData: Function,
+  saveSelectedNode: Function,
+  selectedNode: string
+}
+
+export default class NodeSelect extends React.Component<Props, State> {
+  render() {
+    const { loading, loadNodesData, nodesShown } = this.props
+    return (
+      <FullHeightPanel
+        headerText="Node Selection"
+        renderCloseButton={() => <CloseButton routeTo={ROUTES.SETTINGS} />}
+        renderHeaderIcon={this.renderIcon}
+        renderInstructions={this.renderInstructions}
+        instructionsClassName={styles.instructions}
+      >
+        <section className={styles.tableContainer}>
+          <div className={styles.header}>
+            <Tooltip title="Refresh" className={styles.refresh}>
+              <span onClick={loading ? null : loadNodesData}> Refresh </span>
+              <RefreshIcon
+                id="refresh"
+                onClick={loading ? null : loadNodesData}
+                className={classNames(styles.icon, {
+                  [styles.loading]: loading
+                })}
+              />
+            </Tooltip>
+
+            <div className={styles.count}>Top {nodesShown} nodes listed</div>
+
+            {this.renderAutomaticSelect()}
+          </div>
+          {this.renderNodeList()}
+        </section>
+      </FullHeightPanel>
+    )
+  }
+
+  renderAutomaticSelect = () => {
+    const { selectedNode } = this.props
+    let icon
+
+    if (selectedNode) {
+      icon = (
+        <AddIcon className={styles.icon} onClick={() => this.handleSelect()} />
+      )
+    } else {
+      icon = <ConfirmIcon className={styles.icon} />
+    }
+
+    return (
+      <Tooltip title="Select automatically" className={styles.automaticSelect}>
+        {icon}
+        <span onClick={() => this.handleSelect('')}>Select automatically</span>
```

Comment:
Im slightly confused... Why does select automatically invoke saveSelectedNode with an empty string here?

Reply:
I used an empty string to clear whats saved in storage effectively for the selected node. Should I create a new action the clear the storage instead (to make it clearer)?